packages/ozone/src/mod-service/profile.ts

```diff
@@ -0,0 +1,141 @@
+  'tools.ozone.report.defs#reasonHarassmentHateSpeech': REASONRUDE,
+  'tools.ozone.report.defs#reasonHarassmentDoxxing': REASONRUDE,
+  // Map all misleading reasons to REASONMISLEADING
+  'tools.ozone.report.defs#reasonMisleadingBot': REASONMISLEADING,
+  'tools.ozone.report.defs#reasonMisleadingImpersonation': REASONMISLEADING,
+  'tools.ozone.report.defs#reasonMisleadingScam': REASONMISLEADING,
+  'tools.ozone.report.defs#reasonMisleadingSyntheticContent': REASONMISLEADING,
+  'tools.ozone.report.defs#reasonMisleadingMisinformation': REASONMISLEADING,
+  // Map all rule-related reasons to REASONVIOLATION
+  'tools.ozone.report.defs#reasonRuleSiteSecurity': REASONVIOLATION,
+  'tools.ozone.report.defs#reasonRuleStolenContent': REASONVIOLATION,
+  'tools.ozone.report.defs#reasonRuleProhibitedSales': REASONVIOLATION,
+  'tools.ozone.report.defs#reasonRuleBanEvasion': REASONVIOLATION,
+  // Map all civic reasons to REASONMISLEADING
+  'tools.ozone.report.defs#reasonCivicElectoralProcess': REASONMISLEADING,
+  'tools.ozone.report.defs#reasonCivicDisclosure': REASONMISLEADING,
+  'tools.ozone.report.defs#reasonCivicInterference': REASONMISLEADING,
+  'tools.ozone.report.defs#reasonCivicMisinformation': REASONMISLEADING,
+  'tools.ozone.report.defs#reasonCivicImpersonation': REASONMISLEADING,
+}
+
+interface CacheEntry {
+  profile: AppBskyLabelerDefs.LabelerViewDetailed | null
+  timestamp: number
+}
+
+export type ModerationServiceProfileCreator = () => ModerationServiceProfile
+
+export class ModerationServiceProfile {
+  private cache: CacheEntry | null = null
+  private CACHE_TTL: number
+
+  constructor(
+    private cfg: OzoneConfig,
+    private appviewAgent: AtpAgent,
+    cacheTTL?: number,
+  ) {
+    this.CACHE_TTL = cacheTTL || cfg.service.serviceRecordCacheTTL || 5 * MINUTE
+  }
+
+  static creator(
+    cfg: OzoneConfig,
+    appviewAgent: AtpAgent,
+  ): ModerationServiceProfileCreator {
+    return () => new ModerationServiceProfile(cfg, appviewAgent)
+  }
+
+  async validateReasonType(reasonType: string): Promise<string> {
+    const now = Date.now()
+
+    if (!this.cache || now - this.cache.timestamp > this.CACHE_TTL) {
+      try {
+        const { data } = await this.appviewAgent.app.bsky.labeler.getServices({
+          dids: [this.cfg.service.did],
+          detailed: true,
+        })
+
+        if (AppBskyLabelerDefs.isLabelerViewDetailed(data.views?.[0])) {
+          this.cache = {
+            profile: data.views[0],
+            timestamp: now,
+          }
+        }
+      } catch (e) {
+        // On error, fail open
+        httpLogger.error(`Failed to fetch labeler profile: ${e?.['message']}`)
+      }
+    }
+
+    if (
+      this.cache?.profile?.reasonTypes &&
+      Array.isArray(this.cache.profile.reasonTypes)
+    ) {
+      const supportedReasonTypes = this.cache.profile.reasonTypes
+
+      // Check if the reason type is directly supported
+      if (supportedReasonTypes.includes(reasonType)) {
```

Comment:
Personally, I feel like this entire section can be simplified to:

```suggestion
      const mappedOldReason = NEW_TO_OLD_REASON_MAPPING[reasonType];
      const supportedReasonTypes = this.cache.profile.reasonTypes;

      if (!supportedReasonTypes.includes(reasonType) && 
          !mappedOldReason &&
          !supportedReasonTypes.includes(mappedOldReason)) {
              throw new InvalidRequestError(`Invalid reason type: ${reasonType}`);
      }
      return reasonType;
```

But I also see the point of detailing each part